src/database/migrations/20240815181652-add-permission-for-actions.js

```diff
@@ -0,0 +1,110 @@
+'use strict'
+
+require('module-alias/register')
+const common = require('@constants/common')
+const Permissions = require('@database/models/index').Permission
+
+const getPermissionId = async (module, request_type, api_path) => {
+	try {
+		const permission = await Permissions.findOne({
+			where: { module, request_type, api_path },
+		})
+
+		if (!permission) {
+			throw permission
+		}
+
+		return permission.id
+	} catch (error) {
+		throw error
+	}
+}
+
+module.exports = {
+	up: async (queryInterface, Sequelize) => {
+		try {
+			//create module
+			const modulesData = [
+				{ code: 'actions', status: 'ACTIVE', created_at: new Date(), updated_at: new Date() },
+				{ code: 'activities', status: 'ACTIVE', created_at: new Date(), updated_at: new Date() },
+			]
+
+			// Insert the data into the 'modules' table
+			await queryInterface.bulkInsert('modules', modulesData)
+
+			//create permission
+			const permissionsData = [
+				{
+					code: 'action_permissions',
+					module: 'actions',
+					request_type: ['POST', 'GET', 'DELETE'],
+					api_path: '/scp/v1/actions/*',
+					status: 'ACTIVE',
+					created_at: new Date(),
+					updated_at: new Date(),
+				},
+				{
+					code: 'read_activities_permissions',
+					module: 'activities',
+					request_type: ['GET'],
+					api_path: '/scp/v1/activities/list*',
+					status: 'ACTIVE',
+					created_at: new Date(),
+					updated_at: new Date(),
+				},
+			]
+
+			await queryInterface.bulkInsert('permissions', permissionsData)
+
+			//create role permission mapping
+			const rolePermissionsData = [
+				{
+					role_title: common.ADMIN_ROLE,
+					permission_id: await getPermissionId('actions', ['POST', 'GET', 'DELETE'], '/scp/v1/actions/*'),
+					module: 'actions',
+					request_type: ['POST', 'GET', 'DELETE'],
+					api_path: '/scp/v1/actions/*',
+					created_at: new Date(),
+					updated_at: new Date(),
+					created_by: 0,
+				},
+				{
+					role_title: common.ORG_ADMIN_ROLE,
```

Comment:
Should we give delete / update permission to ORG_ADMIN? Since the activities are not org centric and common for the entire system. @priyanka-TL

Reply:
@adithyadinesh0412  That is not required, its not org centric, but still org admin can create new and enhance or delete one item and instead of that they can add new item. Its there choice, only thing is if they are adding new actions they need to write the code for that

Reply:
@aks30  Having a confusion would like to have your thoughts here.
So here we are giving the permission to ORG_ADMINs to do CRUD operations on Actions table. 
Since actions are not org centric I'm doubting if we have to do that or not. 
My point is; what if an org admin of `A` changes the action description from one language to other or any text change specific to his/her org. It will be reflected throughout the actions in the system which is not right in a SAAS model. 
So should we allow the org admins to do that or should we bring in org concept in actions?